src/operations/generate/java/SpringBootSeed.ts

```diff
@@ -50,14 +52,16 @@ export class SpringBootSeed extends JavaSeed {
      * @param oldClass   name of class to move from
      * @param newClass   name of class to move to
      */
-    protected renameClassStem(project: ProjectNonBlocking, oldClass: string, newClass: string): void {
+    protected renameClassStem<P extends ProjectAsync>(project: P,
+                                                      oldClass: string, newClass: string): Promise<P> {
         logger.info("Replacing old class stem [%s] with [%s]", oldClass, newClass);
-        doWithFiles(project, JavaFiles, f => {
+        return doWithFiles(project, JavaFiles, f => {
```

Comment:
maybe doWithFiles should return the project?

Reply:
or maybe it's returning the modified ones? that would also make sense

Reply:
It's actually the same instance. Like world, it changes